bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productlist/ProductListImpl.java

```diff
@@ -146,6 +163,30 @@ protected void initModel() {
         if (usePlaceholderData) {
             searchResultsSet = new SearchResultsSetImpl();
         } else {
+            Resource fragmentsNode = resource.getChild(ProductList.NN_FRAGMENTS);
+            if (fragmentsNode != null && fragmentsNode.hasChildren()) {
+                Iterable<Resource> configuredFragments = fragmentsNode.getChildren();
+                for (Resource fragment : configuredFragments) {
+                    ValueMap fragmentVm = fragment.getValueMap();
+                    Integer fragmentPage = fragmentVm.get(PN_FRAGMENT_PAGE, Integer.class);
+                    if (fragmentPage.equals(currentPageIndex)) {
```

Comment:
This may produce a null pointer exception. Instead you could use 
`fragmentVm.get(PN_FRAGMENT_PAGE, -1);`  before or do a null check.